core/activity/handler.go

```diff
@@ -5,23 +5,30 @@ import (
 	"github.com/infracloudio/msbotbuilder-go/schema"
 )
 
+// Handler acts as the interface for the client program to define actions on various events from connector service
 type Handler interface {
 	OnMessage(context *TurnContext) (schema.Activity, error)
 }
 
+// HandlerFuncs is an adaptor to let client program specify as many or
+// as few functions to handle events of the connector service while still implementing
+// Handler.
 type HandlerFuncs struct {
 	OnMessageFunc func(turn *TurnContext) (schema.Activity, error)
 }
```

Comment:
We should probably add an example snippet showing how to use handlers.
Something like this: https://godoc.org/k8s.io/client-go/tools/cache#NewInformer

Reply:
Added an example in core/bot_framework_adapter_test.go